plugins/outputs/sensu/sensu.go

```diff
@@ -0,0 +1,468 @@
+
+	return output, err
+}
+
+// Constructs the entity payload
+// Throws when no entity name is provided and fails resolve to hostname
+func (s *Sensu) GetEntity() (*OutputEntity, error) {
+	if s.BackendApiUrl != nil {
+		var entityName string
+		if s.Entity != nil && s.Entity.Name != nil {
+			entityName = *s.Entity.Name
+		} else {
+			defaultHostname, err := os.Hostname()
+
+			if err != nil {
+				return &OutputEntity{}, fmt.Errorf("when resolving hostname")
+			}
+			entityName = defaultHostname
+		}
+
+		return &OutputEntity{
+			Metadata: &OutputMetadata{
+				Name: entityName,
+			},
+		}, nil
+	}
+	return nil, nil
+}
+
+// Constructs the check payload
+// Throws if check name is not provided
+func (s *Sensu) GetCheck(metricPoints []*OutputMetric) (*OutputCheck, error) {
+	count := len(metricPoints)
+
+	if s.Check == nil || s.Check.Name == nil {
+		return &OutputCheck{}, fmt.Errorf("missing check name")
+	}
+
+	return &OutputCheck{
+		Metadata: &OutputMetadata{
+			Name: *s.Check.Name,
+		},
+		Status:               0, // Always OK
+		Issued:               time.Now().Unix(),
+		Output:               "Telegraf agent processed " + strconv.Itoa(count) + " metrics",
+		OutputMetricHandlers: s.GetHandlers(),
+	}, nil
+}
+
+func (s *Sensu) GetHandlers() []string {
+	if s.Metrics == nil || s.Metrics.Handlers == nil {
+		return []string{}
+	}
+	return s.Metrics.Handlers
+}
+
+func getFloat(unk interface{}) (float64, error) {
+	switch i := unk.(type) {
+	case float64:
+		return i, nil
+	case float32:
+		return float64(i), nil
+	case int64:
+		return float64(i), nil
+	case int32:
+		return float64(i), nil
+	case int:
+		return float64(i), nil
+	case uint64:
+		return float64(i), nil
+	case uint32:
+		return float64(i), nil
+	case uint:
+		return float64(i), nil
+	default:
+		return math.NaN(), fmt.Errorf("Non-numeric type could not be converted to float")
+	}
+}
+
+type SensuEntity struct {
+	Name      *string `toml:"name"`
+	Namespace *string `toml:"namespace"`
+}
+
+type SensuCheck struct {
+	Name *string `toml:"name"`
+}
+
+type SensuMetrics struct {
+	Handlers []string `toml:"handlers"`
+}
+
+type OutputMetadata struct {
+	Name string `json:"name"`
+}
+
+type OutputEntity struct {
+	Metadata *OutputMetadata `json:"metadata"`
+}
+
+type OutputCheck struct {
+	Metadata             *OutputMetadata `json:"metadata"`
+	Status               int             `json:"status"`
+	Output               string          `json:"output"`
+	Issued               int64           `json:"issued"`
+	OutputMetricHandlers []string        `json:"output_metric_handlers"`
+}
+
+type OutputMetrics struct {
+	Handlers []string        `json:"handlers"`
+	Metrics  []*OutputMetric `json:"points"`
+}
+
+type OutputMetric struct {
+	Name      string       `json:"name"`
+	Tags      []*OutputTag `json:"tags"`
+	Value     interface{}  `json:"value"`
+	Timestamp int64        `json:"timestamp"`
+}
+
+type OutputTag struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
```

Comment:
Can you please group this with the other struct definitions on top so it's easier to find them as you only need to look at one place.